File: reverseString.py
#reverse a string

def reverseString(s):
    s = list(s)
    x = len(s)-1
    y = int(len(s)/2)
    for i in range(y):
        temp = s[i]
        s[i] = s[x-i]
        s[x-i] = temp
    return "".join(s)
print(reverseString(s))

def reverseString(s):
    newstr = ""
    for i in s:
        newstr = i + newstr
    return newstr

print(reverseString(s))

File: reversestring.js
// --- Directions
// Given a string, return a new string with the reversed
// order of characters
// --- Examples
//   reverse('apple') === 'leppa'
//   reverse('hello') === 'olleh'
//   reverse('Greetings!') === '!sgniteerG'

/*function reverse(str){
    str.split('').reduce();

}*/

/*function reverse(str){
    let reversed = '';
    for(let character of str){
        reversed = character + reversed;
    }
    return reversed;
}*/
//using for loop
function reverse(str) {
    let len = str.length;
    let reversed = '';
    for (i = len; i >=0; i-- ){
       reversed = reversed + str.charAt(i); 
    }
    return reversed;
}
module.exports = reverse;

//using reverse
/*function reverse(str){
    return str.split('').reverse().join('');
}
*/
